gcc (4.9.3) complains:

cc1: warning: command line option '-Wnon-virtual-dtor' is valid for C++/ObjC++ but not for C
cc1: warning: command line option '-Woverloaded-virtual' is valid for C++/ObjC++ but not for C

so set these two flags for CXX only
--- osquery-1.5.0/CMakeLists.txt~	2015-07-20 12:37:24.000000000 +0300
+++ osquery-1.5.0/CMakeLists.txt	2015-07-25 15:06:09.473520609 +0300
@@ -31,10 +31,8 @@
   -Wno-unused-result
   -Wno-missing-field-initializers
   -Wno-sign-compare
-  -Wnon-virtual-dtor
   -Wchar-subscripts
   -Wpointer-arith
-  -Woverloaded-virtual
   -Wformat
   -Wformat-security
   -Werror=format-security
@@ -48,7 +46,7 @@
 
 # osquery additional compiler flags added by CMake.
 set(C_COMPILE_FLAGS "")
-set(CXX_COMPILE_FLAGS "")
+set(CXX_COMPILE_FLAGS "-Wnon-virtual-dtor -Woverloaded-virtual")
 
 # Use osquery language to set platform/os
 if(DEFINED ENV{OSQUERY_PLATFORM})
